main.c

```diff
@@ -111,6 +118,80 @@ struct tstatstruct {
 VEC_STRUCT(tstatsvec,struct tstatstruct);
 #endif
 
+typedef union {
+	u8 raw[pkprefixlen + PUBLIC_LEN + 32];
+	struct {
+		u64 prefix[4];
+		u64 key[4];
+		u64 hash[4];
+	} i;
+} pubonionunion;
+
+
+
+#define BUF_APPEND(buf,offset,src,srclen) strncpy(&buf[offset],src,srclen); offset += srclen;
```

Comment:
Well, besides using do-while construction in an unnatural context and more bloated code, I agree.
Probably it's even better to wrap it in "anonymous" scope, like:
`#define MACROS(...) { /* macros body */ }`
I check with gcc (7.3.0), and it was fine. But I'm not sure what standard says about it and how about support of other compilers.
So it's up to you, what to choose here.

Reply:
there is actually important reason to avoid duplicate `;` -- consider `if (condition) MACRO; else something else;` -- if MACRO expansion results in `{};` or `;;` that'll fuck with else statement